embedded/store/tx.go

```diff
@@ -67,37 +62,21 @@ func NewTxWithEntries(entries []*TxEntry) *Tx {
 	htree, _ := htree.New(len(entries))
 
 	return &Tx{
-		ID:       0,
-		entries:  entries,
-		nentries: len(entries),
-		htree:    htree,
+		header:  &TxHeader{NEntries: len(entries)},
+		entries: entries,
+		htree:   htree,
 	}
 }
 
 func (tx *Tx) Header() *TxHeader {
-	return &TxHeader{
-		ID:       tx.ID,
-		PrevAlh:  tx.PrevAlh,
-		Ts:       tx.Ts,
-		Metadata: tx.Metadata,
-		NEntries: tx.nentries,
-		Eh:       tx.Eh(),
-		BlTxID:   tx.BlTxID,
-		BlRoot:   tx.BlRoot,
-	}
+	return tx.header
```

Comment:
We're returning a pointer to internal struct - that means it can be modified outside (previously it was not possible), is that expected?

Reply:
discussion in the previous comment